### ISSUE TITLE
Hardcoded icons remain despite script running successfully

### ISSUE THREAD

**Issue by acarasimon96** [bilelmoussaoui/Hardcode-Tray#709]
After I ran a system update, all hardcoded icons that were previously overridden reverted back to their originals as if it was done automatically. The hardcoded icons remained the same no matter how many times I ran the script. The last system update did not upgrade all of the apps that the script found, which are listed below.

I tried all SVG converters available on my system (RSVGConvert, Inkscape, CairoSVG, SVGExport) and different sizes, but the hardcoded icons are still not being replaced.

With some investigation, I found out that Hardcode-Tray isn't reading anything from the temp PNG file. The part of the script in question is [line 92 in `HardcodeTray/modules/svg/svg.py`](https://github.com/bilelmoussaoui/Hardcode-Tray/blob/master/HardcodeTray/modules/svg/svg.py#L91). However, the generated temp PNG files are perfectly readable outside Hardcode-Tray (for example, `cat /tmp/tmpxxxxxxxx`).

### Specifications
- Hardcode-Tray version: 4.3 ([Git AUR package](https://aur.archlinux.org/packages/hardcode-tray-git))
- Distribution: Manjaro
- Gtk version: 3.24
- Inkscape version: 1.1
- Cairo version: 1.17.4
- Sni-qt version: not installed
- Icon theme: Papirus
- Python version: 3.9.7
- Linux kernel: 5.14.5-1-MANJARO x86_64
- Apps detected: Chrome (Ungoogled Chromium), Discord, HP Printer, Solaar

[Debug log](https://github.com/bilelmoussaoui/Hardcode-Tray/files/7189338/2021-09-17.22.03.14.log)

**Edit:** Added my Linux kernel version, since this bug deals with temporary files and `/tmp`.

**Comment by varlesh** [comment]
recommend use RSVGConvert.
Try remove backup icons and start script again:
`sudo rm -rf ~/.config/Hardcode-Tray/*`

**Comment by acarasimon96** [comment]
That's what I did every single time before I ran `sudo -E hardcode-tray --apply` again. When I do run `sudo -E hardcode-tray --clear-cache`, everything in that directory is already removed not including itself.

**Comment by varlesh** [comment]
hangouts not support now, because does not pass the test On Chrome/Chromium...
Also all other chrome-based distros need update.
Discord 0.15 not fixed too?

**Comment by varlesh** [comment]
hangouts removed and chrome/chromium updated

**Comment by acarasimon96** [comment]
> Discord 0.15 not fixed too?

Discord has been on 0.15 for a while. Until now, replacing its hardcoded tray icons always worked, and those icons remain even after in-app updates are applied.

**Comment by varlesh** [comment]
it's impossible, discord have hardcoded path with release number for icons:
`{userhome}/.config/discord/0.0.15/modules/discord_desktop_core/`
chrome/chromium & vivaldi updated. See previous commits

**Comment by varlesh** [comment]
I can't reproduce this. 
With latest commits:
before
![Screenshot_20210918_111157](https://user-images.githubusercontent.com/8083855/133879985-c251acc6-b7f6-48a1-81a0-f8cca3a02da4.png)
after
![Screenshot_20210918_111243](https://user-images.githubusercontent.com/8083855/133879979-3fe76c2b-327e-4a6f-84a7-1e103414f28e.png)

**Comment by acarasimon96** [comment]
Does your debug logs still say "Couldn't find PNG file"? I think that's a sign that the script failed to copy the PNG data into the target apps.

**Comment by varlesh** [comment]
yes
[2021-09-18 11:19:30.log](https://github.com/bilelmoussaoui/Hardcode-Tray/files/7189436/2021-09-18.11.19.30.log)

**Comment by varlesh** [comment]
Couldn't find a PNG file - it's previous icon files for previous versions Chrome.
```
    "icons": {
        "tray-NOT-USED-NOW": {
            "original": "8003",
            "theme": "google-chrome-tray"
        },
        "tray-56-NOT-USED-NOW": {
            "original": "10153",
            "theme": "google-chrome-tray"
        },
        "tray-60-dev-NOT-USED-NOW": {
            "original": "10154",
            "theme": "google-chrome-tray"
        },
        "tray-64-NOT-USED-NOW": {
            "original": "10156",
            "theme": "google-chrome-tray"
        },
        "tray-68-NOT-USED-NOW": {
            "original": "10158",
            "theme": "google-chrome-tray"
        },
        "tray-77-NOT-USED-NOW": {
            "original": "10151",
            "theme": "google-chrome-tray"
        },
        "tray-80-NOT-USED-NOW": {
            "original": "9151",
            "theme": "google-chrome-tray"
        },
        "tray-93-NOW-USED": {
            "original": "11151",
            "theme": "google-chrome-tray"
        }
```

**Comment by varlesh** [comment]
![Screenshot_20210918_112939](https://user-images.githubusercontent.com/8083855/133880438-fbc3b036-d363-45df-911f-21402dec2162.png)

**Comment by acarasimon96** [comment]
Sorry for causing you confusion, but I was able to figure out a fix for this strange behavior and replacing icons worked for me like before. This is caused by the temp file object (that is, `temppng` in `SVG.to_bin`) not picking up changes after the PNG conversion. Closing and reopening the file in code allowed the Hardcode Tray to read the contents of the temp PNG. I will submit a pull request that includes this fix and possibly add an exception when the script reads nothing out of the temp file.
